pkg/bffauth/session.go

```diff
@@ -0,0 +1,53 @@
+package bffauth
+
+import (
+	"fmt"
+	"time"
+)
+
+type ChallengeSessionPersister interface {
```

Comment:
I think this should all be internal ... and then we move the specific protocol related pieces out (none of the persisters) into pkg if we want to share with people

Reply:
I changed it to this .. then realized it's possible we'd want to persist this beyond process lifetimes -- but I think we'd still want it to be internal and instead have the constructor take in like "bffAuthPersistedStateFile" -- i.e. the caller should be unaware of implementation details like the fact that this is being persisted.